libafl/src/state/mod.rs

```diff
@@ -58,6 +60,22 @@ pub trait State:
 {
 }
 
+/// The [`State`] of the fuzzer.
+/// Contains all important information about the current run.
+/// Will be used to restart the fuzzing process at any time.
+#[cfg(feature = "std")]
+pub trait State:
+    UsesInput
+    + Serialize
+    + DeserializeOwned
+    + MaybeHasClientPerfMonitor
+    + MaybeHasScalabilityMonitor
+    + HasCurrentCorpusId
+    + HasCurrentStage
+    + HasDumpStateDir
```

Comment:
mmm. can you do this in a way similar to how MaybeHasClientPerfMonitor is implemented?
because if you don't have dump_state feature then the trait name says it has but actually it doesn't have any dir

Reply:
ok makes sense